lib/rucio/core/did.py

```diff
@@ -1290,14 +1290,28 @@ def _delete_dids(
             models.ReplicationRule.rse_expression,
             models.ReplicationRule.locks_ok_cnt,
             models.ReplicationRule.locks_replicating_cnt,
-            models.ReplicationRule.locks_stuck_cnt
-        ).join_from(
+            models.ReplicationRule.locks_stuck_cnt,
+            temp_table.scope,
+            temp_table.name
+        ).outerjoin_from(
             temp_table,
             models.ReplicationRule,
             and_(models.ReplicationRule.scope == temp_table.scope,
                  models.ReplicationRule.name == temp_table.name)
         )
-        for (rule_id, scope, name, rse_expression, locks_ok_cnt, locks_replicating_cnt, locks_stuck_cnt) in session.execute(stmt):
+        for (rule_id, scope, name, rse_expression, locks_ok_cnt, locks_replicating_cnt, locks_stuck_cnt, didscope, didname) in session.execute(stmt):
+            # no rule was found: still need to set tombstone on replicas.
+            if rule_id is None:
+                query = update(
+                    models.RSEFileAssociation
+                ).filter(
+                    models.RSEFileAssociation.scope == didscope,
+                    models.RSEFileAssociation.name == didname
+                ).values(
+                    tombstone=OBSOLETE
+                )
+                session.execute(query)
+                continue
```

Comment:
unnecessary continue? or is it just to skip the logger?

Reply:
necessary continue: this `if` block is reached when no rule could be queried for the given scopename. Some calls in the following code (i.e. `rucio.core.rule.delete_rule(rule_id=rule_id, ...`) will lead to errors when run with `rule_id = None`.

The point you made with the logger is a good one, I added logger calls.